building-and-packaging/material/packaging_debian_notes.md

```diff
@@ -0,0 +1,284 @@
+# Debian packages
+
+## Learning goals
+
+- How to make a CMake project installable.
+- How to package a CMake project via `CPack`.
+- How to create a Debian (`deb`) package of your program/library using CPack.
+- How to check the resulting Debian package.
+
+## Introduction
+
+- Now:
+    - Debian packages for C++ via `CMake`
+    - HPC suitable packages for Python and C++
+        - This requires a well-defined installation routine.
+- Rough plan
+    1. Extend CMake configuration from previous lecture to have an `install` target as well as proper prefix and installation layout (`/bin`, `/include`, `/lib`...)
+    2. Extend CMake configuration to contain a CPack section (rather minimal) such that we can create a package (`tar.gz` file)
+    3. Extend the CMake configuration to create a Debian package
+    4. Check the Debian package with `lintian` and potentially improve the package
+
+## 1. Preparing Project
+
+- Make sure two files are created (a library and en executable). This should be the case when starting with the code from last week's lecture.
+- We add the following changes
+    - Adding a version number to the project. This is not needed yet, but I am afraid of forgetting it later.
+    - We mark the include files needed to use the `libsse` library, i.e., `sse/sse.hpp` publicly visible.
```

Comment:
Why is this necessary?

Reply:
Why is what necessary? Creation of a library and executable or the marking of the headers?

- Regarding the library and executable I have commented on the PR already. 
- Regarding the headers we need to tell CMake what are important headers to be used by external projects. By default all headers are assumed to be used only within the project.

Reply:
I meant the "mark as public".

Reply:
I find [the documentation on this](https://cmake.org/cmake/help/latest/prop_tgt/PUBLIC_HEADER.html) also a bit confusing. It is mentioned there together with some `FRAMEWORK` property which seems to be only relevant for some Apple OSes. However, if I do not set this property in my `CMakeLists.txt`, the include files are not installed.

Reply:
I add an additional remark to the notes and I will try to mention this also in the lecture.